File: scratch/fix_dart_syntax.ts
import fs from "fs";

function fixFile(filePath: string) {
    let content = fs.readFileSync(filePath, "utf8");

    // Replace brandBlue with winGreen
    content = content.replace(/AppColors\.brandBlue/g, "AppColors.winGreen");

    // Fix broken quotes in string interpolation
    content = content.replace(/\\\$\\\{\(p\['/g, "\\\${(p['"); // wait, maybe my node script output exact characters
    
    // Let's just fix it using robust replace
    
    fs.writeFileSync(filePath, content);
}

fixFile("app/lib/screens/dashboard/desktop_dashboard.dart");
fixFile("app/lib/screens/dashboard/mobile_dashboard.dart");

File: scratch/fix_dart_syntax.py
import re

def fix(file):
    with open(file, 'r') as f:
        content = f.read()

    # Find the trade row section and replace all single quotes bounding string interpolation with double quotes
    content = content.replace("'\\${(p[\\'entryPrice\\'] ?? 0).toStringAsFixed(2)}'", '"\\$${(p[\'entryPrice\'] ?? 0).toStringAsFixed(2)}"')
    content = content.replace("'${(p[\\'unrealizedPnl\\'] ?? 0) >= 0 ? '+' : ''}\\${(p[\\'unrealizedPnl\\'] ?? 0).toStringAsFixed(2)}'", '"${(p[\'unrealizedPnl\'] ?? 0) >= 0 ? \'+\' : \'\'}\\$${(p[\'unrealizedPnl\'] ?? 0).toStringAsFixed(2)}"')
    
    with open(file, 'w') as f:
        f.write(content)

fix("app/lib/screens/dashboard/desktop_dashboard.dart")
fix("app/lib/screens/dashboard/mobile_dashboard.dart")
